enaml/core/import_hooks.py

```diff
@@ -368,26 +423,203 @@ def get_code(self):
+        """ Searches for the given Enaml module within a zip and returns an instance
+        of this class on success.  
+        
+        Caching
+        ---------
+        If cache files exist within the archive they are used, otherwise cache files 
+        are created using the standard EnamlImporter's `_write_cache` method. Subsequent 
+        loads be imported by the standard EnamlImporter.
+        
+
+        Paramters
+        ---------
+        fullname : string
+            The fully qualified name of the module.
+
+        path : list or None
+            The subpackage __path__ for submodules and subpackages
+            or None if a top-level module.
+
+        Returns
+        -------
+        results : Instance(AbstractEnamlImporter) or None
+            If the Enaml module is located an instance of the importer
+            that will perform the rest of the operations is returned.
+            Otherwise, returns None.
+
+        """
+        # We're looking inside a package and 'path' the package path
+        if path is not None:
+            modname = fullname.rsplit('.', 1)[-1]
+            pkgpath = fullname.split('.')[:-1]
+            leaf = ''.join((modname, os.path.extsep, 'enaml'))
+            for stem in path:
+                enaml_path = os.path.join(stem, leaf)
+                file_info = make_file_info(enaml_path)
+
+                # Strip package off path to get the archive name
+                archive_path = stem
+                for p in pkgpath:
+                    archive_path = os.path.dirname(archive_path)
+
+                # To check if cache file is in zip file
+                cache_path = os.path.relpath(file_info.cache_path,
+                                             archive_path).replace("\\","/")
+
+                if (cls._is_supported(archive_path) and
+                        os.path.exists(archive_path)):
+
+                    # Path where code should be within the archive
+                    code_path = '/'.join(pkgpath+[leaf])
+                    try:
+                        with ZipFile(archive_path,'r') as archive:
+                            name_list = archive.namelist()
+                            if ((code_path in name_list) or
+                                    (cache_path in name_list)):
+                                return cls(file_info, archive_path)
+                    except IOError:
+                        return
+
+        # We're trying a load a package
+        elif '.' in fullname:
+            return
+
+        # We're doing a direct import
+        else:
+            leaf = fullname + os.path.extsep + 'enaml'
+            for stem in sys.path:
+                enaml_path = os.path.join(stem, leaf)
+                file_info = make_file_info(enaml_path)
+                # To check if cache file is in zip file
+                cache_path = os.path.relpath(file_info.cache_path,
+                                             stem).replace("\\","/")
```

Comment:
Missing space after comma

Reply:
In addition to the ```archive_path```, I would store this and the ```code_path```, on the importer to further reduce code duplication and avoid the need to pass a src_path around.